api/types.go

```diff
@@ -137,29 +137,34 @@ type NodeChainStatus struct {
 	BlocksPerTipsetLastFinality float64
 }
 
-type CheckStatus int
+type CheckStatusCode int
 
-//go:generate go run golang.org/x/tools/cmd/stringer -type=CheckStatus -trimprefix=CheckStatus
+//go:generate go run golang.org/x/tools/cmd/stringer -type=CheckStatusCode -trimprefix=CheckStatus
 const (
-	_ CheckStatus = iota
-	CheckStatusErrSerialize
-	CheckStatusErrTooBig
-	CheckStatusErrInvalid
-	CheckStatusErrMinGas
-	CheckStatusErrMinBaseFee
-	CheckStatusErrBaseFee
-	CheckStatusErrBaseFeeLowerBound
-	CheckStatusErrBaseFeeUpperBound
-	CheckStatusErrGetStateNonce
-	CheckStatusErrBadNonce
-	CheckStatusErrGetStateBalance
-	CheckStatusErrInsufficientBalance
+	_ CheckStatusCode = iota
+	// Message Checks
+	CheckStatusMessageSerialize
+	CheckStatusMessageSize
+	CheckStatusMessageValidity
+	CheckStatusMessageMinGas
+	CheckStatusMessageMinBaseFee
+	CheckStatusMessageBaseFee
+	CheckStatusMessageBaseFeeLowerBound
+	CheckStatusMessageBaseFeeUpperBound
+	CheckStatusMessageGetStateNonce
+	CheckStatusMessageNonce
+	CheckStatusMessageGetStateBalance
+	CheckStatusMessageBalance
 )
 
+type CheckStatus struct {
+	Code CheckStatusCode
+	OK   bool
+	Err  string
+	Hint map[string]interface{}
+}
+
 type MessageCheckStatus struct {
```

Comment:
We need to figure out better names.
I'm thinking about:
for the top-level `CheckResult`
for the message `MessageCheckResult`
for the enum `CheckID`.

Hmm, if we move everything about checks to a separate package then we can strip the `Check` prefix from everything.
So it would be `check.Result`. `check.ID` or `check.Code` and the enum would have shorter prefix.
WDYT?
@vyzo

Reply:
We can certainly do that, I am not fond of comically long names either :)